saga/inMemorySagaLog.go

```diff
@@ -0,0 +1,72 @@
+package saga
+
+import "errors"
+import "fmt"
+import "sync"
+
+/*
+ * In Memory Implementation of a Saga Log, DOES NOT durably persist messages.
+ * This is for local testing purposes.
+ */
+type inMemorySagaLog struct {
+	sagas map[string]*SagaState
+	mutex *sync.RWMutex
```

Comment:
Yes agree.  Actually just talked to Dan about a interface change that will make exposing SagaState Unnecessary.

Also inMemorySaga has locks just because you can concurrently run multiple sagas that are updating the same in memory dictionary concurrently.